src/pages/index.tsx

```diff
@@ -10,7 +10,7 @@ const IndexPage: NextPage = () => {
   const { hasSafes } = useHasSafes()
 
   useEffect(() => {
-    if (!router.isReady || router.pathname !== AppRoutes.index) {
+    if (!router.isReady || router.pathname !== AppRoutes.index || hasSafes === undefined) {
```

Comment:
The issue now is that if I don't have any safes added I will be stuck on the index page. I can't reproduce the issue that Franco found. I have safes added and opening the index page redirects me to the account page and not the welcome page which is what we want.